### PR TITLE
Generalize spinnaker test to work for all devops templates

### PR DESCRIPTION
**Pull request by ejizba**
The generic test:
1. Deploys the template
2. Runs a basic smoke test
3. Uploads correlation id to storage account

### CODE REVIEW

**Comment by ejizba** [comment]
Since git is sometimes bad at picking up file renames, I did the renames in one commit and all my other changes in a second commit. See just the second commit for a better diff for reviewing: https://github.com/azure-devops/ci/pull/15/commits/34a7f196e4992445fe370bff5d851c2d94749e36

File: test-quickstart-template.groovy [hunk 1]
```diff
@@ -4,50 +4,42 @@ node('quickstart-template') {
...
         }
 
-        stage('Deploy Quickstart Template') {
-            def script_path = 'devopsci/scripts/deploy-quickstart-template.sh'
-            sh 'sudo chmod +x ' + script_path
-            withCredentials([usernamePassword(credentialsId: 'AzDevOpsTestingSP', passwordVariable: 'app_key', usernameVariable: 'app_id')]) {
-                withCredentials([string(credentialsId: 'QsStorageConnectionString', variable: 'connection_string')]) {
-                    sh script_path + ' -tn ' + env.JOB_BASE_NAME + ' -sn ' + scenario_name + ' -ai ' + env.app_id + ' -ak ' + env.app_key + ' -cs "' + env.connection_string + '"'
+        if (run_tests) {
+            sh 'ssh -F ' + scenario_name + '/ssh_config -f -N tunnel-start'
```
Comment (clguiman) [inline test-quickstart-template.groovy]: can you run this outside of a stage?

Comment (ejizba) [reply test-quickstart-template.groovy]: I believe so. I'm running Jenkins jobs on my own instance right now to verify these changes and they're not to this point yet

Comment (ejizba) [reply test-quickstart-template.groovy]: I confirmed this runs just fine outside of a stage.

**Review by clguiman — APPROVED** [review]

**Pull request merged**